kythe/cxx/indexer/cxx/GraphObserver.h

```diff
@@ -1174,12 +1179,19 @@ class NullGraphObserver : public GraphObserver {
 
   const ClaimToken* getVNameClaimToken() const override { return &VnameToken; }
 
+  const std::multimap<clang::FileID, std::shared_ptr<MetadataFile>>& meta()
```

Comment:
It seems like this interface wants to be merely:

std::shared_ptr<MetadataFile> GetMetadataFile(clang::SourceRange);

(or CharSourceRange)

Reply:
There can be more than one metadata file associated with a given FileID.

Reply:
Also, `PrepareAlternateSemanticCache` scans through all loaded metadata at once, not just the metadata for a particular source range.